src/Features/CSharp/Portable/NavigationBar/CSharpNavigationBarItemService.cs

```diff
@@ -124,16 +124,15 @@ private static IEnumerable<INamedTypeSymbol> GetTypesInFile(SemanticModel semant
         using (Logger.LogBlock(FunctionId.NavigationBar_ItemService_GetTypesInFile_CSharp, cancellationToken))
         {
             var types = new HashSet<INamedTypeSymbol>();
-            var nodesToVisit = new Stack<SyntaxNode>();
+            using var _ = ArrayBuilder<SyntaxNode>.GetInstance(out var nodesToVisit);
```

Comment:
curious - why are we changing only some usages of `Stack` to `ArrayBuilder`?  Is it instances where we're not using a pooled `Stack`?